src/groupByKey.ts

```diff
@@ -2,6 +2,14 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const groupItems: GroupsMap<T> = {};
+
+  items.forEach((item) => {
+    const itemValue = String(item[key]);
+
+    groupItems[itemValue] = items.filter((obj) => obj[key] === item[key]);
```

Comment:
it is redundsnt to filter items on each iteration.
you can push to existance value or create empty array in case if no exists itemValue key in groupItems.
something like that
```
groupItems[itemValue] = acc[itemValue] || [];
groupItems[itemValue].push(item);
```